src/components/pages/WikiNotePage.tsx

```diff
@@ -461,6 +462,12 @@ const WikiNotePage = ({ storage }: WikiNotePageProps) => {
     topbarTree,
   ])
 
+  useEffect(() => {
+    if (note != null) {
+      addVisitedToHistory(note)
+    }
+  }, [addVisitedToHistory, note])
```

Comment:
We should check id of a note rather than the note object itself. Otherwise, the note will be added again when any attributes of the note are modified since our store updates data immutably

Reply:
Added prevNoteId, does not update it on archiving, tag adding etc, editing. 
Resolved all conflicts as well.